application/helpers/expressions/em_manager_helper.php

```diff
@@ -353,7 +353,8 @@ class LimeExpressionManager
      * 'hidden' => 0    // 1 if it should be always_hidden
      * 'gid' => "34"    // group id
      * 'mandatory' => 'N'   // 'Y' if mandatory, 'S' if soft mandatory
-     * 'eqn' => ""  // TODO ??
+     * 'mandSoftForced' => false // boolean value to keep Mandatroy soft question status. False if not seen, answered, not a soft mandatory or not checked one time. Check is done in self::_validateQuestion using $_POST['mandSoft']
```

Comment:
Thanks! This is more clear now. I would just add something like "If true, validation will fail and user will be asked to confirm the question is not to be answered. After being answered, will be set to false."

Reply:
Seems to be more for the manual ?  https://manual.limesurvey.org/Question_type_-_Array/en#Mandatory

This one need to be updated for "No popup" action.

Reply:
> "Controls when to show a mandatory soft alert to user. 

You can show it in Survey theme where you want, the checkbox is shown currently when try to submit a mandatory question.

> If true, validation will fail and user will be asked to confirm the question is not to be answered.

No : if true user already check the checkbox (or confirm on dialog box)

Reply:
> Seems to be more for the manual ?

For me is documentation for the coder, that's why suggested it there.
But it just a suggestion.
Already approved the code review.

Reply:
>  Already approved the code review.

Yes, i know, but if comment are more clear : it's great :)